src/edu/ucsb/nceas/metacat/dataone/D1AuthHelper.java

```diff
@@ -335,14 +335,30 @@ public void doCNOnlyAuthorization(Session session) throws ServiceFailure, NotAut
 
     /**
      * Does MN/CN admin authorization
-     * @param session
-     * @throws ServiceFailure
-     * @throws NotAuthorized
+     *
+     * @param session A session object that contains a subject value to check for authorization
+     * @throws ServiceFailure When there is an issue checking for authorization
+     * @throws NotAuthorized  When the session subject is not authorized
      */
     public void doAdminAuthorization(Session session) throws ServiceFailure, NotAuthorized {
-        if(session != null && session.getSubject() != null) {
-            logMetacat.debug("D1AuthHepler.doAdminAuthorization - the session is "+session.getSubject().getValue());
+        // First, ensure that the session is valid.
+        if (session == null) {
+            throw new NotAuthorized("0000", "Session is null.");
+        }
+        // If there is no subject or if subject is empty, session is not authorized.
+        String sessionSubject = session.getSubject().getValue();
```

Comment:
If `session.getSubject` is null, the method will get null exception.

Reply:
I have wrapped the relevant block in a try statement to catch this `NullPointerException` and throw a new `NotAuthorized` exception with the message "Session subject is not set."